### DISCUSSION TITLE
[Feature] concurrent uploads for mobile backups

### DISCUSSION THREAD

**Discussion by Ankk98** [immich-app/immich#14183] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

# Feature Proposal: Concurrent File Uploads For Mobile Backups

## Summary
Currently, files are backed up from mobile one at a time. This proposal introduces concurrent uploads, allowing multiple files to be uploaded simultaneously. The number of concurrent uploads can be configured in settings, with a default value of 2.

## This will change these files:
mobile/lib/services/backup.service.dart
mobile/lib/providers/backup/backup.provider.dart
And few other related files for settings, progress etc.

## We can create few new files like:
mobile/lib/models/backup/backup_asset_upload_task.model.dart
mobile/lib/services/backup_asset_uploader.service.dart

Is this a good idea?
If it aligns with the goals of the project, I would like to contribute to it.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Dupe of #9710.

Effort on the mobile upload mechanism is currently probably best spent on refactoring and making it use the right background worker mechanisms, which paves the way for chunked uploading and then potentially parallelizing those chunks. If this is something you're willing to help with please consider joining us on Discord to discuss.
